### ISSUE TITLE
EpoxyController builds models running thread mentioned in Wiki may be wrong.

### ISSUE THREAD

**Issue by Nobler** [airbnb/epoxy#903]
I am reading Wiki about EpoxyController and I find some description may be not correct.
> Note: The first time an EpoxyController builds models it will always run on the main thread, even if you have specified an async handler. 
[link](https://github.com/airbnb/epoxy/wiki/Epoxy-Controller#asynchronous-support)

The running thread of models built in first time should be same as the thread running the `requetModelsBuild`. If it is the first time we call requestModelBuild, buildModelsRunnable will just run in current thread, and buildModels will run in this thread as well.
```
public void requestModelBuild() {
    if (isBuildingModels()) {
      throw new IllegalEpoxyUsage("Cannot call `requestModelBuild` from inside `buildModels`");
    }
    if (hasBuiltModelsEver) {
      requestDelayedModelBuild(0);
    } else {
      buildModelsRunnable.run();
    }
  }
```

Epoxy not check and restrain the running thread of requestModelBuild, if it will always be called in main thread, the Wiki is correct, but it's not.

@elihart Please help confirm that.
Thanks!

**Comment by elihart** [comment]
yes, Epoxy will run its first model build on whichever thread `requestModelBuild` is called from, but it should be an error to not call it from the main thread (although this is not currently checked).

if you don't call it from the main thread synchronously when the view is setup you won't have proper view restoration

**Comment by Nobler** [comment]
In most of the time, view setup and config happend in main thread(in lifecycle of the creation of Activity or Fragment). If a RecyclerView's behavior is not normal which actually don't follow the android specification, Epoxy might need to make some tips.

At least, wiki should give the right instructions.

Anyway, thanks for the reply.
